### PR TITLE
fixed a part of compiler warnings

### PR DESCRIPTION
**Pull request by wlk**
As part of getting familiar with akka-http I went through a part of compiler warnings and fixed them.

Mostly they were related to deprecation, there were few related to dead code.
I have only touched the warnings that are straightforward to fix because I didn't want to break anything

I'm not sure about the warnings related to `parboiled2` - should they be fixed here, or in the https://github.com/sirthias/parboiled2 repository ?

There was no issue related to this PR

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Thanks, very cool effort :-)
Yeah, correct - parboiled we want to leave as is, because it's a mirror of the public one. We do this in order no to lock in people in the parboiled dependency.

**Comment by wlk** [comment]
ok, I'll revert my changes to parboiled and will contribute fixes directly there, also looking at the failure

**Comment by ktoso** [comment]
Great, thanks :)

-- 
Konrad Malawski

From: Wojciech Langiewicz <notifications@github.com>
<notifications@github.com>
Reply: akka/akka-http
<reply@reply.github.com>
<reply@reply.github.com>
Date: 17 September 2016 at 09:09:31
To: akka/akka-http <akka-http@noreply.github.com>
<akka-http@noreply.github.com>
CC: Konrad Malawski <konrad.malawski@lightbend.com>
<konrad.malawski@lightbend.com>, Comment <comment@noreply.github.com>
<comment@noreply.github.com>
Subject:  Re: [akka/akka-http] fixed a part of compiler warnings (#328)

ok, I'll revert my changes to parboiled and will contribute fixes directly

**Comment by wlk** [comment]
parboiled changes reverted, test failure fixed

**Comment by wlk** [comment]
This is the failure:
```
[info] - should open a second connection if the first one is loaded *** FAILED ***
[info]   java.lang.AssertionError: Expected OnNext(_), yet no element signaled during 3 seconds
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:268)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:259)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$TestSetup.acceptIncomingConnection(ConnectionPoolSpec.scala:431)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$2$$anon$5.<init>(ConnectionPoolSpec.scala:81)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(ConnectionPoolSpec.scala:72)
[info]   at akka.http.impl.engine.client.ConnectionPoolSpec$$anonfun$1$$anonfun$apply$mcV$sp$2.apply(ConnectionPoolSpec.scala:72)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   ...
```
Not sure what to do about this, because it's passing locally

**Comment by wlk** [comment]
After re-running the test they passed

**Comment by ktoso** [comment]
Please address Johannes's comments, otherwise LGTM :)

File: akka-http-core/src/test/scala/akka/testkit/AkkaSpec.scala [hunk 1]
```diff
@@ -106,13 +106,13 @@ abstract class AkkaSpec(_system: ActorSystem)
...
-    new Constraint[Class[A], Class[B]] {
+  implicit def classEqualityConstraint[A, B]: CanEqual[Class[A], Class[B]] =
+    new CanEqual[Class[A], Class[B]] {
       def areEqual(a: Class[A], b: Class[B]) = a == b
     }
 
-  implicit def setEqualityConstraint[A, T <: Set[_ <: A]]: Constraint[Set[A], T] =
-    new Constraint[Set[A], T] {
+  implicit def setEqualityConstraint[A, T <: Set[_ <: A]]: CanEqual[Set[A], T] =
+    new CanEqual[Set[A], T] {
```
Comment (wlk) [inline akka-http-core/src/test/scala/akka/testkit/AkkaSpec.scala]: Yes, it's from the scalatest update:
> Renamed Constraint to CanEqual and made a deprecated type alias for any user code currently referring to Constraint

File: akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala [hunk 2]
```diff
@@ -402,7 +402,7 @@ class ConnectionPoolSpec extends AkkaSpec("""
     val (serverEndpoint, serverHostName, serverPort) = TestUtils.temporaryServerHostnameAndPort()
 
     def testServerHandler(connNr: Int): HttpRequest ⇒ HttpResponse = {
-      case r: HttpRequest ⇒ HttpResponse(headers = responseHeaders(r, connNr), entity = r.entity)
+      r: HttpRequest ⇒ HttpResponse(headers = responseHeaders(r, connNr), entity = r.entity)
```
Comment (wlk) [inline akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala]: fixed

File: akka-http-core/src/test/scala/akka/http/scaladsl/settings/SettingsEqualitySpec.scala [hunk 3]
```diff
@@ -11,17 +11,18 @@ import org.scalatest.WordSpec
...
-      file-get-conditional = on
-      render-vanity-footer = yes
-      range-coalescing-threshold = 80
-      range-count-limit = 16
-      decode-max-bytes-per-chunk = 1m
-      file-io-dispatcher = ${akka.stream.blocking-io-dispatcher}
-    }
-  """).withFallback(ConfigFactory.load).resolve
+  val config = ConfigFactory.parseString(
+    """
```
Comment (wlk) [inline akka-http-core/src/test/scala/akka/http/scaladsl/settings/SettingsEqualitySpec.scala]: fixed

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/TcpLeakApp.scala [hunk 4]
```diff
@@ -35,12 +36,12 @@ object TcpLeakApp extends App {
         .toMat(Sink.head)(Keep.right).run())
     .last
     .onComplete {
-      case error ⇒
+      error ⇒
```
Comment (wlk) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/TcpLeakApp.scala]: I think it was just expected this demo app to fail, renaming to "result" makes sense IMO

**Comment by wlk** [comment]
pushing again to rerun the tests, not sure why but some tests can fail from time to time

**Comment by wlk** [comment]
After rerunning tests passed, do you require squashing commits?

File: akka-http-core/src/test/scala/akka/testkit/AkkaSpec.scala (hunk 1 above)
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/testkit/AkkaSpec.scala]: Cool, thanks for the explanation.

File: akka-http-tests/src/test/scala/akka/http/scaladsl/server/TcpLeakApp.scala [hunk 5]
```diff
@@ -35,12 +36,13 @@ object TcpLeakApp extends App {
         .toMat(Sink.head)(Keep.right).run())
     .last
     .onComplete {
-      case error ⇒
-        println(s"Error: $error")
+      result ⇒
```
Comment (jrudolph) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/server/TcpLeakApp.scala]: Thanks, :+1:

**Comment by jrudolph** [comment]
Thanks again, @wlk. I squashed your commits directly from the PR.

**Comment by wlk** [comment]
Thanks, I'll pick this up again soon and fix another batch of warnings

**Pull request merged**